src/pointcloud_to_laserscan_nodelet.cpp

```diff
@@ -142,7 +158,17 @@ namespace pointcloud_to_laserscan
   void PointCloudToLaserScanNodelet::disconnectCb()
   {
     boost::mutex::scoped_lock lock(connect_mutex_);
-    int num_subscribers = marking_pub_.getNumSubscribers() + clearing_pub_.getNumSubscribers();
+    int num_subscribers = 0;
+    if (use_marking_scan_)
+    {
+      num_subscribers += marking_pub_.getNumSubscribers();
```

Comment:
Shouldn't need to check use_marking_scan_ since the number of subscribers should be correct.

Reply:
If use_XXX_scan is false, I don't initiate the publisher at all, and I believe this will crash? (Not test yet)

Reply:
Should be safe to call whether it is initialized or not (http://docs.ros.org/groovy/api/roscpp/html/publisher_8cpp_source.html)